### ISSUE TITLE
Dynamically loaded asset images from custom widget not loading in FF Test Mode

### ISSUE THREAD

**Issue by jeffgalak** [FlutterFlow/flutterflow-issues#2271]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Custom widget that calls an image asset doesn't load in FF Test Mode, even though it does work in deployment on iOS and Android.

### Expected Behavior

The image should load in FF Test Mode.

### Steps to Reproduce

1. Create new project.
2. Upload any image into the asset library.
3. Create a custom widget with this code replacing "return Container()

return Image.asset(
      "assets/images/IMAGE NAME.jpg",
    );

where "IMAGE NAME.jpg" is the name of the image file that was uploaded in Stop 2.

1. Place custom widget on page.
2. Run app in Test Mode

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT4Ci/Lqw412sbQD1pSJKe9VmSQWCDt/RY080dV9cEwYN5PND+x/esnRQFJuZ9Ordglqf1qjvz4z+tbQm9z2UMApAz6aGLU5wZNIVDzgXjiibpDUCJGWRXZEGshPGkiwy6bUvyMlPrNuWiQs2Dm1N/KibA/dJpjHDFQdD/+hLpLXrgfuQ0SLb3kNh1JWeC/v

### Context

I'm trying to load asset images dynamically using code. Since FF doesn't allow dynamic asset image selection natively in the image widget, I created the custom widget explained aboe.

### Visual documentation

![Screenshot 2024-01-30 at 7 54 35 PM](https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/d4f816db-b61e-481c-a4fd-1134d27540d0/bcf451b7-546b-45a4-b2f5-ff36dd3bdb07?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS9kNGY4MTZkYi1iNjFlLTQ4MWMtYTRmZC0xMTM0ZDI3NTQwZDAvYmNmNDUxYjctNTQ2Yi00NWE0LWIyZjUtZmYzNmRkM2JkYjA3IiwiaWF0IjoxNzA2NjY5NTgzLCJleHAiOjMzMjc3MjI5NTgzfQ.xxtXe_xBoIy_9ehgaEY33XytZD_yER0t64mO5b4zwLQ)

The upper image is loaded directly using the image widget (not-dynamic). The lower failure to load is the custom widget that calls the exact same image (with the same path) but fails to show it. Note that this DOES work in a live deployment in iOS and Android, so this is a problem localized to FF Test Mode.

### Additional Info

*No response*

### Environment

```
- FlutterFlow version: 4.1
- Platform: FF Test Mode
- Browser name and version: Chrome, Safari
- Operating system and version affected: Mac Sonoma 14.1.2
```

**Comment by hariprasadms** [comment]
Hi @jeffgalak - Thank you for reporting the issue. In order to better assist you, could you please provide more details about the image link that is not displaying through the custom widget? Specifically:

Where is the image hosted?
Have you tried using a different image link as a test (e.g., use a link like https://dummyimage.com/600x400)?
Have you encountered any CORS blocking issues when trying to access images? check this error in browser console.
Share the image link details that are not showing up in test mode.

This information will help in investigating the issue and finding a resolution. Additionally, how does the image display in run mode? Please provide these details for further assistance.

**Comment by jeffgalak** [comment]
Sure. I don't actually know where the images are hosted since they are
inserted into FF via the "assets" page. For the test app I set up for this
I never set up Firebase, though that is what I use elsewhere.

I have tried many different images. The issue only comes up with loading
images, from a custom widget in the FF Test view. Again, it loads just fine
on an actual iOS or Android device. No CORS issues that I'm aware of it.

I'm not sure what to share. You;re welcome to access the FF app directly.
The project ID is "markdown-t-e-s-t-p7e48k"

Thank you,
Jeff
Jeff Galak
he/him (what's this?
<https://theconversation.com/what-are-gender-pronouns-and-why-is-it-important-to-use-the-right-ones-169025>
)
